app/views/jobs/edit.html.erb

```diff
@@ -1,3 +1,6 @@
-<h1> Edit Job</h1>
-
-<%= render partial:"job_partial", locals:{job: @job} %>
\ No newline at end of file
+<% if user_signed_in? %>
+  <h1> Edit Job</h1>
+  <%= render partial: "job", locals: { job: @job } %>
+<% else %>
+  <% controller.redirect_to root_path %>
```

Comment:
Well done in finding a solution for this. 

This works, but it's not ideal. The logic we have (knowing when, how and where to redirect a user if they are not logged in) shouldn't be the responsibility of the view. The responsibility of the view is to show things to the user. 

We can refactor this in a few different ways. Personally I would move this logic to our controller using:

```
before_action :authenticate_user!, only: [:edit]
```

Doing this will redirect to the login page. You can leave it as this (the behaviour makes sense to me) but there's also a way to redirect to the home page by overriding the default behaviour if you want to (but definitely optional, happy for you to leave the default behaviour).

Reply:
i'm assuming that the authenticate_user method is defined in devise as a default, or is this something that you're just aware of as you get more experience

Reply:
refactored 👍

Reply:
It's a devise method - see [docs](https://github.com/heartcombo/devise#controller-filters-and-helpers)